### PR TITLE
Try to assert the message was processed via contract state

### LINKED ISSUE #922: Improve Kathy's delivery detection
Currently we have been seeing a lot of block skew errors that seem to be the result of providers switching our connection between instances in the backend and leading to a 1000+ block shift suddenly. This causes ethers-js to not notify us if the event we were watching for was emitted within one of the blocks that was jumped past.

A solution to this is to add an additional query to see if the event has been submitted to the inbox contract at the timeout point as a final check to prevent false error reports.

One thing we should keep in mind is that there is a memory leak here because the `.once` that is used to watch for the event will never be triggered and over time the number of filters being run will increase. I am not sure if we should worry about this at this time, but generally JS does not have a way to cancel promises so we have been potentially leaking as it stands for any of the query that never actually resolve.

### PR DESCRIPTION
**Pull request by nambrot**
Fixes #922 

This PR tries to assert message processing via the stats stored on the HelloWorld contract

### CODE REVIEW

File: typescript/infra/scripts/helloworld/kathy.ts [hunk 1]
```diff
@@ -250,11 +251,20 @@ async function sendMessage(
...
+  try {
+    await utils.timeout(
+      app.waitForMessageReceipt(receipt),
+      MESSAGE_RECEIPT_TIMEOUT,
+      'Timeout waiting for message to be received',
+    );
+  } catch (error) {
+    // If we weren't able to get the receipt for message processing, try to read the state to ensure it wasn't a transient provider issue
+    const statsAfter = await app.channelStats(origin, destination);
+    if (statsAfter.received === statsBefore.received) {
```
Comment (mattiecnvr) [inline typescript/infra/scripts/helloworld/kathy.ts:263]: This comparison does not feel right to me, can you elaborate on why this works? (Not sure what the stats object looks like but concerned about deep equality issues as well)

Comment (nambrot) [reply typescript/infra/scripts/helloworld/kathy.ts:263]: The stats come from the on-chain state here https://github.com/abacus-network/abacus-monorepo/blob/2312953c204c57c34792ece472853e6dd4cfd1b6/typescript/helloworld/contracts/HelloWorld.sol#L19

They get incremented upon processing of the message on the receiving side. Thus when a message has not been processed, the two numbers will be the same. If the message has been processed, the statsAfter received number should be 1 higher than before

Comment (mattiecnvr) [inline typescript/infra/scripts/helloworld/kathy.ts:263]:
```suggestion
    if (channelStatsNow.received <= channelStatsBefore.received) {
      log("Did not receive event for message delivery even though it was delivered", {origin, destination})
```

File: typescript/infra/scripts/helloworld/kathy.ts [hunk 2]
```diff
@@ -250,11 +251,20 @@ async function sendMessage(
...
-  );
+  try {
+    await utils.timeout(
+      app.waitForMessageReceipt(receipt),
+      MESSAGE_RECEIPT_TIMEOUT,
+      'Timeout waiting for message to be received',
+    );
+  } catch (error) {
+    // If we weren't able to get the receipt for message processing, try to read the state to ensure it wasn't a transient provider issue
+    const statsAfter = await app.channelStats(origin, destination);
```
Comment (mattiecnvr) [inline typescript/infra/scripts/helloworld/kathy.ts:262]:
```suggestion
    const channelStatsNow = await app.channelStats(origin, destination);
```

File: typescript/infra/scripts/helloworld/kathy.ts [hunk 3]
```diff
@@ -237,6 +237,7 @@ async function sendMessage(
   // Log it as an obvious reminder
   log('Intentionally setting interchain gas payment to 1');
 
+  const statsBefore = await app.channelStats(origin, destination);
```
Comment (mattiecnvr) [inline typescript/infra/scripts/helloworld/kathy.ts:240]:
```suggestion
  const channelStatsBefore = await app.channelStats(origin, destination);
```

File: typescript/infra/scripts/helloworld/kathy.ts (hunk 1 above)
Comment (mattiecnvr) [reply typescript/infra/scripts/helloworld/kathy.ts:263]: I realize `<` should be impossible but it also makes it more clear when reading that it is a numeric comparison.

**Review by mattiecnvr — APPROVED** [review]

**Pull request merged**